plugins/jobtype/test/azkaban/jobtype/TestHadoopSecureSparkWrapper.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright 2012 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package azkaban.jobtype;
+
+import java.lang.reflect.Field;
+import java.net.URL;
+import java.util.Collections;
+import java.util.Map;
+import java.util.TreeMap;
+
+import org.apache.log4j.Logger;
+import org.apache.hadoop.conf.Configuration;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class TestHadoopSecureSparkWrapper {
+  private static Logger logger = Logger.getRootLogger();
+
+  @Test
+  public void testAutoLabeling() {
```

Comment:
This test covers the code path for both this added feature as well as the one introduced to enforce node labeling setting. It makes sure that with proper configuration, Spark job type manipulates user supplied or system default Spark submit parameters correctly to ensure the job is submitted to the right queue with the right label.